apps/emails-and-messages/src/modules/event-handlers/send-event-messages.ts

```diff
@@ -73,16 +74,30 @@ export const sendEventMessages = async ({
       }
     }
   }
-  const sendgridStatus = await sendSendgrid({
-    authData,
-    channel,
-    event,
-    payload,
-    recipientEmail,
-  });
 
-  if (sendgridStatus?.errors.length) {
-    logger.error("Sending message with Sendgrid has failed");
-    logger.error(sendgridStatus?.errors);
+  if (channelAppConfiguration.sendgridConfigurationId) {
+    logger.debug("Channel has assigned Sendgrid configuration");
+
+    const sendgridConfigurationService = new SendgridConfigurationService({
+      apiClient: client,
+      saleorApiUrl: authData.saleorApiUrl,
+    });
+
+    const sendgridConfiguration = await sendgridConfigurationService.getConfiguration({
+      id: channelAppConfiguration.sendgridConfigurationId,
+    });
+    if (sendgridConfiguration) {
```

Comment:
If provider is not configured, we are passing it to the next one. So it's intentional.

In the next version I'll separate the events per provider and turn it on/off based on configuration instead of static subscriptions created at app install.